tools/functional-tester/etcd-runner/main.go

```diff
@@ -212,6 +219,166 @@ func runRacer(eps []string, round int) {
 	doRounds(rcs, round)
 }
 
+func runWatcher(eps []string) {
+	var runningTime time.Duration = 30 // time for which operation should be performed
+	noOfPrefixes := 36                 // total number of prefixes which will be watched upon
+	watchPerPrefix := 10               // number of watchers per prefix
+	reqRate := 20                      // put request per second
+	keyPrePrefix := 100000             // max number of keyPrePrefixs for put operation
+
+	ctx := context.Background()
+	prefixKeysMap := make(map[string][]string) // map for holding prefixes and values
+	// pre generate prefixes equal to noOfPrefixes
+	prefixes := generateUniqueKeys(5, noOfPrefixes)
+	// pre generate keys
+	keys := generateRandomKeys(10, keyPrePrefix)
+
+	var (
+		revision int64
+		wg       sync.WaitGroup
+		pr       *clientv3.PutResponse
+		gr       *clientv3.GetResponse
+		err      error
+	)
+
+	for i := range prefixes {
+		prefix := prefixes[i]
+		// initialize prefix key map for each prefix
+		prefixKeysMap[prefix] = make([]string, 0)
```

Comment:
@sinsharat 

Will add a `-` char between prefix and key works for you? Basically change my original example code to 

```
  go func() {
        for i := 0; i < len(keys); i++ {
            for j := 0; j < len(prefixes); j++ {
                key := prefixes[j] + "-" + keys[i]

                if err = limiter.Wait(ctxt); err != nil {
                    break
                }

                _, err = client.Put(ctxt, key, key)
                if err == context.DeadlineExceeded {
                    break
                }

                if err != nil {
                    log.Printf("Error: %v occured while trying to key: %v, value : %v to kv store.", err, key, key)
                    continue
                }
            }
        }
    }()
```

we also need to change the watch part.